pyiceberg/expressions/parser.py

```diff
@@ -233,10 +233,14 @@ def handle_not(result: ParseResults) -> Not:
 
 
 def handle_and(result: ParseResults) -> And:
+    if len(result[0]) > 2:
+        return And(result[0][0], result[0][1], *result[0][2:])
     return And(result[0][0], result[0][1])
```

Comment:
Uh maybe not the most pythonic...I guess a ternary which sets the *args option if it's > 2 otherwise AlwaysTrue (for And) and then for Or would be AlwaysFalse so it just considers the first 2.

Reply:
I realized I can just pass in *result[0]